File: SmartPocketAPI/Models/RecurringPayment.cs
using System.Text.Json.Serialization;

namespace SmartPocketAPI.Models;

public class RecurringPayment
{
    public int Id { get; set; } = 0;
    public string Description { get; set; } = string.Empty;
    public bool IsInterestFreePayment { get; set; }
    public int InstallmentCount { get; set; }
    public int NextInstallmentCount { get; set; } = 1;
    public decimal InstallmentAmount { get; set; }
    public decimal InstallmentAmountPerPeriod { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public DateOnly NextInstallmentDate { get; set; }  
    public DateOnly? LastInstallmentDate { get; set; }
    public bool IsActive { get; set; } = true;

    public int CategoryId { get; set; }
    public Category Category { get; set; }

    public int PaymentMethodId { get; set; }
    public PaymentMethod PaymentMethod { get; set; }

    public int MovementTypeId { get; set; }
    public MovementType MovementType { get; set; }

    public int? CreditCardPaymentId { get; set; }
    public PaymentMethod CreditCardPayment { get; set; }

    public Guid UserId { get; set; }
    public User User { get; set; }

    public int FrequencyId { get; set; }
    public Frequency Frequency { get; set; }

    [JsonIgnore]
    public ICollection<Movement> Movements { get; set; }

}

File: SmartPocketWeb2/src/app/models/movements/RecurringPayment.ts
import { User } from "../auth/User";
import { Frequency } from "./Frequency";
import { Category } from "./Category";
import { PaymentMethod } from "./PaymentMethod";
import { MovementType } from "./MovementType";


export interface RecurringPayment {
    Id: number;
    Description: string;
    IsInterestFreePayment: boolean;
    InstallmentCount: number;
    NextInstallmentCount: number;
    InstallmentAmount: number;
    InstallmentAmountPerPeriod: number;
    StartDate: Date;
    EndDate?: Date;
    NextInstallmentDate: Date;
    LastInstallmentDate?: Date;
    IsActive: boolean;
    CategoryId: number;
    Category: Category;
    PaymentMethodId: number;
    PaymentMethod: PaymentMethod;
    MovementTypeId: number;
    MovementType: MovementType;
    CreditCardPaymentId?: number;
    CreditCardPayment: PaymentMethod;
    UserId: string;
    User: User;
    FrequencyId: number;
    Frequency: Frequency;
    CreatedAt: Date;
    UpdatedAt: Date;
}
